src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Oid.cs

```diff
@@ -86,17 +96,40 @@ public static Oid FromOidValue(string oidValue, OidGroup group)
             }
             set
             {
-                _friendlyName = value;
+                // If _friendlyName has not been set, permit it to be set once, or to
+                // the same value for "initialize once" behavior.
+                if (_friendlyName != null && !_friendlyName.Equals(value, StringComparison.Ordinal))
+                {
+                    throw new PlatformNotSupportedException(SR.Cryptography_Oid_SetOnceFriendlyName);
+                }
+
                 // If we can find the matching OID value, then update it as well
-                if (_friendlyName != null)
+                if (value != null)
```

Comment:
I think the better check here is `if (_friendlyName == null)`, so that assigning the current value doesn't re-rerun this logic.

Reply:
As written I think this fails:

```C#
Oid oid = new Oid(SHA1_Value, "SHA256");
// This will throw, because it wants to change _value
oid.FriendlyName = oid.FriendlyName;
```

Whether you take the change to make that not throw or not, there should be a test codifying the behavior.

Reply:
You're right, and I don't think it should throw. Seems to defeat the point of allowing writing a non-mutating value. Fixed.